owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/security/PassCodeViewModel.kt

```diff
@@ -0,0 +1,59 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Juan Carlos Garrote Gascón
+ *
+ * Copyright (C) 2021 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.viewmodels.security
+
+import androidx.lifecycle.ViewModel
+import com.owncloud.android.data.preferences.datasources.SharedPreferencesProvider
+import com.owncloud.android.presentation.ui.security.PassCodeActivity
+
+class PassCodeViewModel(
+    private val preferencesProvider: SharedPreferencesProvider
+) : ViewModel() {
+
+    fun setPassCode(passcode: String) {
+        preferencesProvider.putString(PassCodeActivity.PREFERENCE_PASSCODE, passcode)
+        preferencesProvider.putBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, true)
+    }
+
+    fun removePassCode() {
+        preferencesProvider.putBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
+    }
+
+    fun checkPassCodeIsValid(passCodeDigits: Array<String?>): Boolean {
+        val passcodeString = preferencesProvider.getString(PassCodeActivity.PREFERENCE_PASSCODE, loadPinFromOldFormatIfPossible())
+        var isValid = true
+        var i = 0
+        while (i < passCodeDigits.size && isValid) {
+            val originalDigit = passcodeString!![i].toString()
```

Comment:
Try to avoid using `!!` here. In case `passcodeString` is `null`, we can return `isValid = false` directly, can't we? Same when `PREFERENCE_PASSCODE` is `null` and `loadPinFromOldFormatIfPossible` returns an empty string

Reply:
Ok, I added a pair of tests to check these situations